pandas/io/formats/style_render.py

```diff
@@ -515,7 +515,12 @@ def _generate_col_header_row(self, iter: tuple, max_cols: int, col_lengths: dict
 
         return index_blanks + column_name + column_headers
 
-    def _generate_index_names_row(self, iter: tuple, max_cols: int, col_lengths: dict):
+    def _generate_index_names_row(
+        self,
+        iter: tuple | list,
```

Comment:
Can you narrow these down further?

Reply:
The typehint on the `iter` paramater says its expecting a tuple, but the only time this function its used a list of tuples is being used, so i think this typehint should be changed

Reply:
I meant something like tuple[a, b] with the appropriate types passed in.

Also we are trying to type functions in the way that all possibly valid arguments can be passed in

Reply:
OK, will look into it and do it as soon as i can, thx